packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -696,7 +728,24 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         const builder = new SummaryTreeBuilder();
-        builder.addBlob(`${gcBlobPrefix}_root`, JSON.stringify(gcState));
+        const sortedGCState: IGarbageCollectionState = generateSortedGCState(gcState);
+        const newBlob = JSON.stringify(sortedGCState);
+
+        if (
+            this.latestSerializedGCBlobs.get(gcBlobRootKey) === newBlob &&
+            this.latestSerializedGCBlobs.size === 1 &&
+            this.trackGCBlobState &&
+            !fullTree &&
+            trackState
+        ) {
+            const gcSummaryPath = `/${gcTreeKey}/${gcBlobRootKey}`;
+            builder.addHandle(gcBlobRootKey, SummaryType.Blob, gcSummaryPath);
+            return builder.getSummaryTree();
+        }
+
+        this.pendingSerializedGCBlobs = new Map();
+        this.pendingSerializedGCBlobs.set(gcBlobRootKey, newBlob);
```

Comment:
The pending state should always be set when summarize is called irrespective of whether a handle is sent or not. There can be scenarios where the pending state is not the same as the current state and the next latest state would then be incorrect.